superdesk/sequences.py

```diff
@@ -1,37 +1,91 @@
+import superdesk
 from superdesk import get_resource_service
+from .resource import Resource
+from .services import BaseService
+from flask import current_app as app
 
 
-def get_next_sequence_number(
-    resource_name, item_id,
-    max_seq_number, min_seq_number=0,
-    field_name='sequence_number'
-):
-    """
-    Generates Sequence Number
-    :param resource_name: target resource
-    :param item_id: target item id
-    :param max_seq_number: maximal possible value
-    :param min_seq_number: default 0, minimal possible value
-    :param field_name: default 'sequence_number', name of field to store sequence_number in the target item
-    :returns: sequence_number
-    """
-
-    target_resource = get_resource_service(resource_name)
-    sequence_number = target_resource.find_one(
-        req=None,
-        _id=item_id
-    ).get(field_name) or min_seq_number
-    if (min_seq_number <= sequence_number + 1 <= max_seq_number):
-        target_resource.find_and_modify(
-            query={'_id': item_id},
-            update={'$inc': {field_name: 1}},
-            upsert=False
-        )
-    else:
-        target_resource.find_and_modify(
-            query={'_id': item_id},
-            update={field_name: min_seq_number},
-            upsert=False
-        )
+def init_app(app):
+    endpoint_name = 'sequences'
+    service = SequencesService(endpoint_name, backend=superdesk.get_backend())
+    SequencesResource(endpoint_name, app=app, service=service)
+
+
+class SequencesResource(Resource):
+    schema = {
+        'name': {
+            'type': 'string',
+            'required': True,
+            'nullable': False,
+            'empty': False,
+            'iunique': True
+        },
+        'sequence_number': {
+            'type': 'number',
+            'default': 0,
+        },
+    }
+    etag_ignore_fields = ['sequence_number', 'name']
+    internal_resource = True
+
+
+class SequencesService(BaseService):
 
-    return sequence_number
+    def get_next_sequence_number_for_item(
```

Comment:
For consistency all sequences should be stored in the sequences collection, there's no need for customizing the collection where the sequence should be stored. This is also error prone. This would simplify the API to:
`def get_next_sequence_number(self, key_name, max_seq_no=None, min_seq_no=0)`